config/routes.rb

```diff
@@ -1,11 +1,17 @@
 Rails.application.routes.draw do
+  resources :roles
+
   root 'welcome#index'
 
-  devise_for :users, controllers: { registrations: 'registrations' }
+  devise_for :users, controllers: { registrations: 'registrations' },
+             path_names: { sign_in: 'login', sign_out: 'logout'}
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.